### PR TITLE
cxxopts: Add limits header

### PR DESCRIPTION
**Pull request by kraj**
needed for numeric_limits

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by hunyadi-dev** [comment]
Can you please open a Jira and explain what error did you encounter following the format of other commits on the project?

File: libminifi/src/utils/StringUtils.cpp [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 #include "utils/Environment.h"
 
+#include <limits>
```
Comment (hunyadi-dev) [inline libminifi/src/utils/StringUtils.cpp:22]: `<limits>` should be already included in this chain: `StringUtils.h -> gsl.h -> gsl-lite.hpp`.

Comment (kraj) [reply libminifi/src/utils/StringUtils.cpp:22]: @hunyadi-dev this is seen when compiling with upcoming GCC 11

**Comment by kraj** [comment]
> Can you please open a Jira and explain what error did you encounter following the format of other commits on the project?

I dont have JIRA account etc. and I am not a regular contributor/developer for this component. this was more for awareness for the devs to fix this if the pull is not sufficient, feel free to ignore it.

**Comment by adamdebreceni** [comment]
thank you for your contribution, I have created a jira issue to track this problem [MINIFICPP-1523](https://issues.apache.org/jira/browse/MINIFICPP-1523)

**Comment by arpadboda** [comment]
@kraj : please fix the linter error:

```
/Users/runner/work/nifi-minifi-cpp/nifi-minifi-cpp/libminifi/src//utils/StringUtils.cpp:22:  Found C++ system header after other header. Should be: StringUtils.h, c system, c++ system, other.  [build/include_order] [4]
```

You can verify by running "make linter"

**Review by fgerlits — APPROVED** [review]

**Comment by arpadboda** [comment]
@kraj : thanks for fixing this!

You can subscribe to nifi dev list ( https://nifi.apache.org/mailing_lists.html ) and request contributor access there. 
After doing so, you can create and assign tickets to yourself.

**Pull request closed without merging**